macros/shared/src/main/scala/zio/prelude/Assertion.scala

```diff
@@ -360,8 +372,8 @@ object Assertion {
       def compile: String = s"$char"
     }
 
-    final case class CharacterSet(set: Set[Char], reversed: Boolean) extends Regex {
-      def compile: String = set.mkString(if (reversed) "[^" else "[", "", "]")
+    final case class CharacterSet(set: Set[Regex], reversed: Boolean) extends Regex {
```

Comment:
Modified `CharacterSet` so it accepts `Set[Regex]` instead of `Set[Char]` for better composability